XenAdmin/Controls/Wlb/WlbOptModeScheduler.cs

```diff
@@ -646,8 +646,17 @@ private void buttonDelete_Click(object sender, EventArgs e)
         {
             if (lvTaskList.SelectedItems.Count > 0)
             {
-                WlbScheduledTask task = TaskFromItem(lvTaskList.SelectedItems[0]);
-                DeleteTask(task);
+                var confirmResult = MessageBox.Show(this,
+                                                    Messages.DELETE_WLB_OPTIMIZATION_SCHEDULE_WARNING,
+                                                    Messages.DELETE_WLB_OPTIMIZATION_SCHEDULE_CAPTION,
+                                                    MessageBoxButtons.YesNo,
+                                                    MessageBoxIcon.Warning, MessageBoxDefaultButton.Button1);
+
+                if (confirmResult == DialogResult.Yes)
+                {
+                    WlbScheduledTask task = TaskFromItem(lvTaskList.SelectedItems[0]);
+                    DeleteTask(task);
+                }

```

Comment:
I know this not part of your change, but I believe that the `weekView1.Refresh()` call should be inside the `if` block, as there is no need to refresh if nothing has chanhed.